packages/docs/docs/02-Guides/04-Testing.mdx

```diff
@@ -1,35 +1,90 @@
 # Testing
 
-Let's take ``useTokenAllowance`` as an example.
+Let's take ``useTokenAllowance`` hook as an example.
 
-To write a test, start with a setup code that will create a mock provider and test wallets.
+## Setup
+
+First of all we'll have to setup local ethereum node and create a provider for it. We should also have access to some accounts on this network with some funds on them. All this can be easily using the [Waffle](https://github.com/TrueFiEng/Waffle) testing library.
 
 ```tsx
+  import { MockProvider } from 'ethereum-waffle'
+
   const mockProvider = new MockProvider()
   const [deployer, spender] = mockProvider.getWallets()
 ```
 
-Before each test, deploy an ERC20 contract. It's important as otherwise the result of one
-test could break the other one.
+Next we need to deploy all the neccesary contracts and create a ``useDApp`` config object. ``useDApp`` requires ``Multicall`` contract to be deployed in order to be able to query blockchain. ``@usedapp/core`` package is shipped together with the ``Multicall`` object, that contains all the neccesary data for deployment.
+
+```tsx
+  import { MultiCall } from '@usedapp/core'
+
+  const multicall = await deployContract(deployer, Multicall)
```

Comment:
Is it possible to leverage multicall auto-deployment mechanics and do not trouble the user with it?

Reply:
Valid point. Will take a look into it later.